pkg/kv/kvserver/replica_raft.go

```diff
@@ -767,9 +770,9 @@ func (r *Replica) handleRaftReadyRaftMuLocked(
 ) (stats handleRaftReadyStats, _ error) {
 	// handleRaftReadyRaftMuLocked is not prepared to handle context cancellation,
 	// so assert that it's given a non-cancellable context.
-	if ctx.Done() != nil {
-		return handleRaftReadyStats{}, errors.AssertionFailedf(
-			"handleRaftReadyRaftMuLocked cannot be called with a cancellable context")
+	if err := must.Zero(ctx, ctx.Done(),
```

Comment:
Looks like we need a "nilable" constraint in the upstream package :)

Reply:
Maybe we could enumerate these things manually, to have a constraint that matches all the nilable things. Haven't researched whether the syntax allows for that.

Reply:
I do think must.Zero is a surprising developer UX to test nil interface references.

Would it be possible to introduce an alias must.NilInterface or NilReference and recommend its use for that case?

Reply:
Yes, but we'd need one for each reference type, so about 6 in total, plus the NotNil variants, so 12. I've been hoping to find a way to finagle generics into doing it instead.

Reply:
Interfaces are also tricky, because we'd have to handle typed nils -- I'm not sure we can without reflection. But there's always the `must.True(ctx, foo == nil)` escape hatch too. I agree though, we should try to do something better here.